scimath_f/lawson.f

```diff
@@ -855,20 +905,22 @@ SUBROUTINE DNNLS (A,MDA,M,N,B,X,RNORM,W,ZZ,INDEX,ITMAX,MODE)
 C
 C         COMPUTE COMPONENTS OF THE DUAL (NEGATIVE GRADIENT) VECTOR W().
 C
-          DO 50 IZ=IZ1,IZ2
+        DO IZ=IZ1,IZ2
           J=INDEX(IZ)
           SM=ZERO
-              DO 40 L=NPP1,M
-   40         SM=SM+A(L,J)*B(L)
-   50     W(J)=SM
+            DO L=NPP1,M
+              SM=SM+A(L,J)*B(L)
+            END DO
+          W(J)=SM
+        END DO
 C                                   FIND LARGEST POSITIVE W(J).
    60 WMAX=ZERO
-          DO 70 IZ=IZ1,IZ2
+        DO 70 IZ=IZ1,IZ2
           J=INDEX(IZ)
           IF (W(J).LE.WMAX) GO TO 70
           WMAX=W(J)
           IZMAX=IZ
-   70     CONTINUE
+   70   CONTINUE
```

Comment:
I think this should be `END DO`. The `CONTINUE` statement is a no-op.

Reply:
Ah wait, you did not convert this DO-loop to the line-number free variant. So it should be fine.